mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -168,6 +174,30 @@ def Rock_GemmOp :
   }];
 }
 
+
+def Rock_ReduceOp :
+  Rock_Op<"reduce", [AllElementTypesMatch<["in", "out"]>]>,
+  Arguments<(ins
+    Arg<TensorOrMemRefOf<[F32]>, "input", [MemRead]>:$in,
+    Arg<TensorOrMemRefOf<[F32]>, "output", [MemRead, MemWrite]>:$out,
+    ReduceMethodAttr:$reduceMethod,
+    IndexAttr:$axis,
+    I32Attr:$blockSize,
+    I32Attr:$gridSize,
+    OptionalAttr<BoolAttr>:$useLDS,
```

Comment:
This is perhaps more nicely spelled `UnitAttr`

... also, note to past me, `OptionalAttr<UnitAttr> === UnitAttr` and so I should've not done that.

Reply:
Done!